app/webpacker/controllers/cookie-acceptance_controller.js

```diff
@@ -0,0 +1,43 @@
+import CookiePreferences from "../javascript/cookie_preferences"
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = ["banner", "accept"];
+
+  connect() {
+    if (!this.isPreferencesPage()) {
+      this.checkforCookie();
+    }
+  }
+
+  checkforCookie() {
+    const cookiePrefs = new CookiePreferences() ;
+    if (cookiePrefs.cookieSet)
+      return ;
+
+    this.showBanner();
+  }
+
+  allowAll(event) {
+    event.preventDefault();
+    this.element.classList.add('accepted') ;
+
+    const cookiePrefs = new CookiePreferences() ;
+    cookiePrefs.allowAll() ;
+  }
+
+  showBanner() {
+    this.element.classList.remove('hide') ;
+    this.acceptTarget.focus() ;
+  }
+
+  isPreferencesPage() {
+    const path = window.location.href.replace(/^https?:\/\/[^/]+/, '')
```

Comment:
`window.location.pathname` will give you this I think

Reply:
Thats better will switch to that

Reply:
In the future! will merge for now